pkg/apis/core/validation/validation.go

```diff
@@ -1503,6 +1513,39 @@ func validateCSIPersistentVolumeSource(csi *core.CSIPersistentVolumeSource, fldP
 	return allErrs
 }
 
+func validateCSIVolumeSource(csi *core.CSIVolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(csi.Driver) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("driver"), ""))
+	} else {
+		if len(csi.Driver) > 63 {
```

Comment:
Can the shared fields between CSIVolumeSource and CSIPersistentVolumeSource use a shared function so we make sure validation between the two are the same?

Reply:
See response in newer PR here - https://github.com/kubernetes/kubernetes/pull/74086#issuecomment-464188838